### PR TITLE
[pickers] Add `localeText` prop to all pickers to customize the translations

### LINKED ISSUE #5844: [pickers] Allow to override some local texts directly on a picker component
### Order ID 💳

46586

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

Hello,

We're using the new DateRangePicker with LocalizationProvider but are now encountering some unexpected behavior regarding labeling.

We want to be able to label different pickers with different labels depending on their form context. For example one will have a start label "Absence from" while another might only have "From" or "Start at".

From what I understand, this currently required multiple LocalizationProviders as it's where the labels currently are defined. See https://mui.com/x/react-date-pickers/localization/

Is there a way to do this without defining multiple LocalizationProviders or it that the way to go? It feels counter intuitive to us at the moment.

Our preferred way of doing this would be to use the `startText` and `endText` props but those are deprecated. See https://mui.com/x/api/date-pickers/date-range-picker/#props

Thank you for your help on this issue.

Best regards
Olivier

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```  System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.7.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Browsers:
    Chrome: 104.0.5112.101
    Firefox: 103.0.2
  npmPackages:
    @emotion/react: ^11.7.1 => 11.9.3 
    @emotion/styled: ^11.6.0 => 11.9.3 
    @mui/base:  5.0.0-alpha.86 
    @mui/icons-material: ^5.2.5 => 5.8.4 
    @mui/lab: ^5.0.0-alpha.54 => 5.0.0-alpha.87 
    @mui/material: ^5.2.7 => 5.8.5 
    @mui/private-theming:  5.8.4 
    @mui/styled-engine:  5.8.0 
    @mui/styles: ^5.2.3 => 5.8.4 
    @mui/system: ^5.2.6 => 5.8.5 
    @mui/types:  7.1.4 
    @mui/utils:  5.8.4 
    @mui/x-date-pickers:  5.0.0-beta.1 
    @mui/x-date-pickers-pro: ^5.0.0-beta.1 => 5.0.0-beta.1 
    @mui/x-license-pro:  5.12.1 
    @types/react: ^17.0.38 => 17.0.47 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    typescript: ^4.4.4 => 4.7.4 
```
</details>

### PR DESCRIPTION
**Pull request by flaviendelangle**
Closes #5844
[Doc preview](https://deploy-preview-6212--material-ui-x.netlify.app/x/react-date-pickers/localization/#text-translations)

The `localeText` allows to customize some translations on all the the basic pickers (Static, Mobile and Desktop variants of DatePicker, DateTimePicker, TimePicker and DateRangePicker).
It will also be added to the fields in the future if they have translations.
Should we also add it to the "views" (CalendarPicker for instance) ?

## Changes

- Remove the `startText` / `endText` props because the locale fallback was applied outside of the new context so the new prop was not applied
- Add a `localeText` prop to all pickers

### CODE REVIEW

**Review by alexfauquette — APPROVED** [review]
Sound good, I assume you let the 3 remaining `deprecatedPropsWarning` to be removed for another PR

**Comment by flaviendelangle** [comment]
> Sound good, I assume you let the 3 remaining deprecatedPropsWarning to be removed for another PR

Yes, 2 in https://github.com/mui/mui-x/pull/5890 and the last one in a follow up PR.

File: docs/data/date-pickers/localization/localization.md [hunk 1]
```diff
@@ -114,7 +120,15 @@ import bgLocale from 'date-fns/locale/bg';
 </LocalizationProvider>;
 ```
 
-## Supported locales
+### Using the component prop
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/date-pickers/localization/localization.md:123]: Good point :+1:

File: docs/data/date-pickers/localization/localization.md [hunk 2]
```diff
@@ -34,13 +34,16 @@ In the example bellow, we add a dot at the end of each day in the calendar heade
...
 
 As the rest of MUI components, you can modify text and translations.
 You can find all the translation keys supported in [the source](https://github.com/mui/mui-x/blob/HEAD/packages/x-date-pickers/src/locales/utils/pickersLocaleTextApi.ts)
 in the GitHub repository.
 
-You can set the locale text by using the theme provider.
+### Using the theme
+
+To translate all your components from `@mui/x-date-pickers` and `@mui/x-date-pickers-pro`,
+you just have to import the locale from `@mui/x-date-pikers` (see the [list of supported locales below](/#supported-locales)).
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/date-pickers/localization/localization.md:46]: I just removed the `/` to have a valid anchor.

**Pull request merged**